runtime/polkadot/tests/weights.rs

```diff
@@ -0,0 +1,347 @@
+}
+
+#[test]
+fn sanity_check_weight_per_nanos_is_as_expected() {
+	// This value comes from Substrate, we want to make sure that if it
+	// ever changes we don't accidently break Polkadot
+	assert_eq!(WEIGHT_PER_NANOS, 1_000)
+}
+
+#[test]
+fn weight_of_balances_transfer_is_correct() {
+	// #[weight = T::DbWeight::get().reads_writes(1, 1) + 70_000_000]
+	let expected_weight = 195_000_000;
+
+	let weight = polkadot_runtime::BalancesCall::transfer::<Runtime>(Default::default(), Default::default())
+		.get_dispatch_info()
+		.weight;
+	assert_eq!(weight, expected_weight);
+}
+
+#[test]
+fn weight_of_balances_set_balance_is_correct() {
+	// #[weight = T::DbWeight::get().reads_writes(1, 1) + 35_000_000]
+	let expected_weight = 160_000_000;
+
+	let weight = polkadot_runtime::BalancesCall::set_balance::<Runtime>(
+		Default::default(),
+		Default::default(),
+		Default::default(),
+	)
+	.get_dispatch_info()
+	.weight;
+
+	assert_eq!(weight, expected_weight);
+}
+
+#[test]
+fn weight_of_balances_force_transfer_is_correct() {
+	// #[weight = T::DbWeight::get().reads_writes(2, 2) + 70_000_000]
+	let expected_weight = 320_000_000;
+
+	let weight = polkadot_runtime::BalancesCall::force_transfer::<Runtime>(
+		Default::default(),
+		Default::default(),
+		Default::default(),
+	)
+	.get_dispatch_info()
+	.weight;
+
+	assert_eq!(weight, expected_weight);
+}
+
+#[test]
+fn weight_of_balances_transfer_keep_alive_is_correct() {
+	// #[weight = T::DbWeight::get().reads_writes(1, 1) + 50_000_000]
+	let expected_weight = 175_000_000;
+
+	let weight = polkadot_runtime::BalancesCall::transfer_keep_alive::<Runtime>(Default::default(), Default::default())
+		.get_dispatch_info()
+		.weight;
+
+	assert_eq!(weight, expected_weight);
+}
+
+#[test]
+fn weight_of_timestap_set_is_correct() {
+	// #[weight = T::DbWeight::get().reads_writes(2, 1) + 9_000_000]
+	let expected_weight = 159_000_000;
+	let weight = polkadot_runtime::TimestampCall::set::<Runtime>(Default::default()).get_dispatch_info().weight;
+
+	assert_eq!(weight, expected_weight);
+}
+
+#[test]
+fn weight_of_staking_bond_is_correct() {
+	let controller: AccountId = AccountKeyring::Alice.into();
+
+	// #[weight = 500_000_000]
+	let expected_weight = 500_000_000;
+	let weight = StakingCall::bond::<Runtime>(controller, 1 * DOLLARS, Default::default()).get_dispatch_info().weight;
+
+	assert_eq!(weight, expected_weight);
+}
+
+#[test]
+fn weight_of_staking_bond_extra_is_correct() {
+	// #[weight = 500_000_000]
+	let expected_weight = 500_000_000;
+	let weight = StakingCall::bond_extra::<Runtime>(1 * DOLLARS).get_dispatch_info().weight;
+
+	assert_eq!(weight, expected_weight);
+}
+
+#[test]
+fn weight_of_staking_unbond_is_correct() {
+	// #[weight = 400_000_000]
+	let expected_weight = 400_000_000;
+	let weight = StakingCall::unbond::<Runtime>(Default::default()).get_dispatch_info().weight;
+
+	assert_eq!(weight, expected_weight);
+}
+
+#[test]
+fn weight_of_staking_widthdraw_unbonded_is_correct() {
```

Comment:
a lot of these values will be changing coming up i.e. staking

Reply:
Yeah, which forces me to ask: what type of _breakage_ is this preventing? 

- I like the `sanity_checks_...` even though the chance of breaking them is low. 
- As for the rest, I think anything that breaks one of them, will break all of them, so I think only checking balance transfer is enough.

Reply:
I disagree. Changes in one Pallet don't necessarily affect another Pallet. Tests can and should evolve over time, and I don't think these tests are particularly challenging to maintain.

I think the advantage of having these tests is that it makes sure that people realize a change happened in Substrate (by seeing a companion PR to update the tests) instead of silently having weights changed from under them.

In an extreme example, I guess we'd want to prevent something like EIP 1884, which changed the pricing of certain OPCODEs and thus some [security assumptions](https://github.com/holiman/eip-1884-security).

Reply:
> instead of silently having weights changed from under them.

Makes sense.